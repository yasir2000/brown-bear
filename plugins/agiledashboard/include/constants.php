<?php
/**
 * Copyright (c) Enalean, 2012 - Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

define('AGILEDASHBOARD_BASE_URL', '/plugins/agiledashboard');
define('AGILEDASHBOARD_BASE_DIR', dirname(__FILE__));
define('AGILEDASHBOARD_TEMPLATE_DIR', dirname(__FILE__) . '/../templates');

/**
 * Get the pane to display on agile dashboard index page
 *
 * Parameters:
 * 'milestone'         => The current Planning_Milestone
 * 'user'              => The current user
 * 'milestone_factory' => Planning_MilestoneFactory
 *
 * Expected results
 * 'pane'        => Input/Output parameter, and AgileDashboard_Pane
 */
define('AGILEDASHBOARD_EVENT_INDEX_PAGE', 'agiledashboard_event_index_page');

/**
 * Modify the redirect parameters when attempt to display a planning without specific Milestone selected
 *
 * Parameters:
 * 'milestone' => The most recent Planning_Milestone on which we are about to be redirected
 *
 * Expected results
 * 'redirect_parameters' => Input/Output parameter, array of 'key' => 'value'
 */
define('AGILEDASHBOARD_EVENT_MILESTONE_SELECTOR_REDIRECT', 'agiledashboard_event_milestone_selector_redirect');

/**
 * Fetch the cardwall configuration html
 *
 * Parameters:
 * 'tracker' => The Planning Tracker of the planning that is being configured
 * 'view'    => The HTML to be fetched
 */
define('AGILEDASHBOARD_EVENT_PLANNING_CONFIG', 'agiledashboard_event_planning_config');

/**
 * Update a planning
 *
 * Parameters:
 * 'tracker' => The Planning Tracker of the planning that is being configured
 * 'request' => The standard request object
 */
define('AGILEDASHBOARD_EVENT_PLANNING_CONFIG_UPDATE', 'agiledashboard_event_planning_config_update');

/**
 * RESt call for cardwall GET
 *
 * Parameters:
 * 'version'   => API version
 * 'milestone' => Milestone on which cardwall is requested
 */
define('AGILEDASHBOARD_EVENT_REST_GET_CARDWALL', 'agiledashboard_event_rest_get_cardwall');

/**
 * RESt call for burndown GET
 *
 * Parameters:
 * 'version'   => API version
 * 'user'      => The user who resquest the burndown
 * 'milestone' => Milestone on which burndown is requested
 * 'burndown'  => OUT the \Tuleap\Tracker\REST\Artifact\BurndownRepresentation
 */
define('AGILEDASHBOARD_EVENT_REST_GET_BURNDOWN', 'agiledashboard_event_rest_get_burndown');

/**
 * RESt call for burndown OPTIONS
 *
 * Parameters:
 * 'version'   => API version
 * 'user'      => The user who resquest the burndown
 * 'milestone' => Milestone on which burndown is requested
 */
define('AGILEDASHBOARD_EVENT_REST_OPTIONS_BURNDOWN', 'agiledashboard_event_rest_options_burndown');


/**
 * Allow plugins to modify the milestone requested by a RESt call
 *
 * Parameters:
 * 'user'                                       => The user who requested
 * 'milestone'                                  => The Milestone object
 * 'milestone_representation_reference_holder' => The reference to a RESt representation of the milestone
 */
define('AGILEDASHBOARD_EVENT_REST_GET_MILESTONE', 'agiledashboard_event_rest_get_milestone');

/**
 * Checks if cardwall is enabled
 *
 * Parameters:
 * 'tracker' => The Planning Tracker of the planning that is being configured
 * 'enabled' => boolean
 */
define('AGILEDASHBOARD_EVENT_IS_CARDWALL_ENABLED', 'agiledashboard_event_is_cardwall_enabled');

/**
 * Get backlog item tracker card fields semantic
 *
 * Parameters:
 * 'tracker'              => The Tracker of the backlog item
 * 'card_fields_semantic' => Tracker card fields semantic
 */
define('AGILEDASHBOARD_EVENT_GET_CARD_FIELDS', 'agiledashboard_event_get_card_fields');

/**
 * Get Cardwall REST resources only if agiledashboard is activated
 */
define('AGILEDASHBOARD_EVENT_REST_RESOURCES', 'agiledashboard_event_rest_resources');

/**
 * Export AD structure in XML
 */
define('AGILEDASHBOARD_EXPORT_XML', 'agiledashboard_export_xml');
