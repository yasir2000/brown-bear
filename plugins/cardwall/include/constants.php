<?php
/**
 * Copyright (c) Enalean, 2012 - Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

define('CARDWALL_BASE_DIR', dirname(__FILE__));
define('CARDWALL_BASE_URL', '/plugins/cardwall');

/**
 * Can we use standard javascript
 *
 * Parameters:
 *   'use_standard' => boolean
 *
 */
define('CARDWALL_EVENT_USE_STANDARD_JAVASCRIPT', 'cardwall_event_use_standard_javascript');
