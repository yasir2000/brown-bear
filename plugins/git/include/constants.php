<?php
/**
 * Copyright (c) Enalean, 2012 - Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

define('GIT_BASE_URL', '/plugins/git');
define('GIT_SITE_ADMIN_BASE_URL', '/admin/git/');
define('GIT_BASE_DIR', dirname(__FILE__));
define('GIT_TEMPLATE_DIR', GIT_BASE_DIR . '/../templates');
define('GITOLITE3_LOGS_PATH', '/var/lib/gitolite/.gitolite/logs/');

/**
 * Check if platform can use gerrit
 *
 * Parameters:
 *     'platform_can_use_gerrit' => boolean
 */
define('GIT_EVENT_PLATFORM_CAN_USE_GERRIT', 'git_event_platform_can_use_gerrit');
define('REST_GIT_PULL_REQUEST_ENDPOINTS', 'rest_git_pull_request_endpoints');
define('REST_GIT_PULL_REQUEST_GET_FOR_REPOSITORY', 'rest_git_pull_request_get_for_repository');

/**
 * Allow a plugin to append his own classes to the body DOM element in git views
 *
 * Parameters:
 *   'request' => (Input)  Codendi_Request Request
 *   'classes' => (Output) String[]        Additional classnames
 */
define('GIT_ADDITIONAL_BODY_CLASSES', 'git_additional_body_classes');

/**
 * Allow a plugin to add permitted git actions
 *
 * Parameters:
 *   'repository'        => (Input)  GitRepository Git repository
 *   'user'              => (Input)  PFUser        Current user
 *   'permitted_actions' => (Output) String[]      Permitted actions
 */
define('GIT_ADDITIONAL_PERMITTED_ACTIONS', 'git_additional_permitted_actions');

/**
 * Allow plugins to add additional notifications setup for git
 *
 * Parameters:
 *   'repository' => (Input) GitRepository Git repository currently modified
 *   'request'    => (Input) HTTPRequest   Current request
 *   'output'     => (Output) String       The HTML to present
 */
define('GIT_ADDITIONAL_NOTIFICATIONS', 'git_additional_notifications');
