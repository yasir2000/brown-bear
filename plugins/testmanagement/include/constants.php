<?php
/**
 * Copyright (c) Enalean, 2014 - Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

define('TESTMANAGEMENT_BASE_URL', '/plugins/testmanagement');
define('TESTMANAGEMENT_BASE_DIR', realpath(__DIR__ . '/..'));
define('TESTMANAGEMENT_RESOURCE_DIR', TESTMANAGEMENT_BASE_DIR . '/resources');
define('TESTMANAGEMENT_GETTEXT_DIR', TESTMANAGEMENT_BASE_DIR . '/site-content');
define('CAMPAIGN_TRACKER_SHORTNAME', 'campaign');
define('DEFINITION_TRACKER_SHORTNAME', 'test_def');
define('EXECUTION_TRACKER_SHORTNAME', 'test_exec');
define('ISSUE_TRACKER_SHORTNAME', 'bug');
