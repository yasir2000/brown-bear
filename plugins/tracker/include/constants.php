<?php
/**
 * Copyright (c) Enalean, 2012 - Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */
define('TRACKER_BASE_URL', '/plugins/tracker');
define('TRACKER_BASE_DIR', dirname(__FILE__));
define('TRACKER_TEMPLATE_DIR', realpath(dirname(__FILE__) . '/../templates'));

define('TRACKER_EVENT_INCLUDE_CSS_FILE', 'tracker_event_include_css_file');

/**
 * The trackers from a project have been duplicated in another project
 *
 * Parameters:
 * 'tracker_mapping'   => The mapping between source and target project trackers
 * 'field_mapping'     => The mapping between source and target fields
 * 'group_id'          => The id of the target project
 * 'ugroups_mapping'   => The mapping between source and target ugroups
 * 'source_project_id' => The id of the source project
 *
 * No expected results
 */
define('TRACKER_EVENT_TRACKERS_DUPLICATED', 'tracker_event_trackers_duplicated');

/**
 * An artifact has just been (un)associated to another one
 *
 * Parameters:
 * 'artifact'             => The artifact which receive the (un)association
 * 'linked-artifact-id'   => The (previously) linked artifact id
 * 'request'              => The request
 * 'user'                 => The user who made the request
 * 'form_element_factory' => The FormElementFactory
 *
 * Expected results:
 *  No expected results
 */
define('TRACKER_EVENT_ARTIFACT_ASSOCIATION_EDITED', 'tracker_event_artifact_association_edited');

/**
 * Request a custom type from other plugins for a new artifact link
 *
 * Parameters:
 * 'project_id'      => The id of the target project
 * 'to_artifact'     => The artifact linked to
 * 'submitted_value' => Values from the artifact form
 *
 * Expected results:
 * 'nature'          => string the type proposed by the plugin
 */
define('TRACKER_EVENT_ARTIFACT_LINK_TYPE_REQUESTED', 'tracker_event_artifact_link_type_requested');

/**
 * Fetch the semantics used by other plugins
 *
 * Parameters:
 * 'semantics' => @var Tracker_SemanticCollection A collection of semantics that needs adding to.
 * 'tracker'   => @var Tracker                    The Tracker the semantics are defined upon
 *
 * Expected results
 * The semantics parameter is populated with additional semantic fields
 */
define('TRACKER_EVENT_MANAGE_SEMANTICS', 'tracker_event_manage_semantics');

/**
 * Create a semantic from xml in other plugins
 *
 * Parameters:
 * 'xml'           => @var SimpleXMLElement
 * 'xml_mapping'   => @var array
 * 'tracker'       => @var Tracker
 * 'semantic'      => @var array
 * 'type'          => @var string
 *
 * Expected results
 * The semantic parameter is populated with a Tracker_Semantic object if it exists for the given type
 */
define('TRACKER_EVENT_SEMANTIC_FROM_XML', 'tracker_event_semantic_from_xml');

/**
 * Fetches all the semantic names
 *
 * Parameters:
 * 'semantic' => @var array of semantic name strings
 */
define('TRACKER_EVENT_GET_SEMANTICS_NAMES', 'tracker_event_get_semantics_names');

/**
 * Get the various duplicators that can duplicate semantics
 *
 * Parameters:
 *  'duplicators' => \Tuleap\Tracker\Semantic\IDuplicateSemantic[]
 */
define('TRACKER_EVENT_GET_SEMANTIC_DUPLICATORS', 'tracker_event_get_semantic_duplicators');

/**
 * Get the various criteria that may enhance a report
 *
 * Parameters:
 *  'array_of_html_criteria' string[]                (OUT) html code to be included in the criteria list
 *  'tracker'                Tracker                 (IN)  the current tracker
 *  'additional_criteria'    Tracker_Report_AdditionalCriteria[]  (IN)
 *  'user'                   PFUser                  (IN)  the current user
 */
define('TRACKER_EVENT_REPORT_DISPLAY_ADDITIONAL_CRITERIA', 'tracker_event_report_display_additional_criteria');

/**
 * We want to save in database additional criteria
 *
 * Parameters:
 * 'additional_criteria'    Tracker_Report_AdditionalCriteria[]  (IN)
 * 'report'                 Tracker_Report                       (IN)
 */
define('TRACKER_EVENT_REPORT_SAVE_ADDITIONAL_CRITERIA', 'tracker_event_report_save_additional_criteria');

/**
 * We want to save in database additional criteria
 *
 * Parameters:
 * 'additional_criteria_values'    array($key => $value) (OUT)
 * 'report'                        Tracker_Report        (IN)
 */
define('TRACKER_EVENT_REPORT_LOAD_ADDITIONAL_CRITERIA', 'tracker_event_report_load_additional_criteria');

/**
 * Event emitted when a field data can be augmented by plugins
 *
 * Parameters:
 *   'additional_criteria'    Tracker_Report_AdditionalCriteria[]  (IN)
 *   'result'                 String (OUT)
 *   'artifact_id'            Int (IN)
 *   'field'                  Tracker_FormElement_Field (IN)
 */
define('TRACKER_EVENT_FIELD_AUGMENT_DATA_FOR_REPORT', 'tracker_event_field_augment_data_for_report');

/**
 * Event emitted to check if a tracker can be deleted
 *
 * Parameters:
 *   'tracker'                Tracker (IN)
 *   'result'                 Array (OUT)
 */
define('TRACKER_USAGE', 'tracker_usage');

/**
 * Event emitted to check if a user can change the priority of an artifact
 *
 * Parameters:
 *  'user_is_authorized'    BOOl    (OUT)
 *  'group_id'              INT     (IN)
 *  'milestone_id'          IN      (IN)
 *  'user'                  PFUser  (IN)
 */
define('ITEM_PRIORITY_CHANGE', 'item_priority_change');

/**
 * Event emitted to delete tracker
 *
 * Parameters:
 *  'tracker_id'      int (IN)
 *  'key'             string  (IN)
 */
define('TRACKER_EVENT_DELETE_TRACKER', 'tracker_event_delete_tracker');

/**
 * Event emitted to display tracker admin buttons
 *
 * Parameters:
 *  'tracker_id'      int (IN)
 */
define('TRACKER_EVENT_FETCH_ADMIN_BUTTONS', 'tracker_event_fetch_admin_buttons');

/**
 * Getting informations from agile dashboard plugin about the tracker affilitaion to an agile dashboard
 *
 * Parameters:
 *  'cannot_configure_instantiate_for_new_projects' Boolean
 *  'tracker'                                       Tracker
 */
define('TRACKER_EVENT_GENERAL_SETTINGS', 'tracker_event_general_settings');

/**
 * Get the trackers required by agile dashboard
 *
 * Parameters:
 *  'project_id'        project_id
 *  'tracker_ids_list'  array containing tracker ids
*/
define('TRACKER_EVENT_PROJECT_CREATION_TRACKERS_REQUIRED', 'tracker_event_project_creation_trackers_required');
