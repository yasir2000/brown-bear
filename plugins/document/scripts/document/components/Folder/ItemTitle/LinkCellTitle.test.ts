/*
 * Copyright (c) Enalean, 2019 - Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 *
 */

import LinkCellTitle from "./LinkCellTitle.vue";
import { createLocalVue, shallowMount } from "@vue/test-utils";
import { TYPE_LINK } from "../../../constants";
import { createStoreMock } from "@tuleap/core/scripts/vue-components/store-wrapper-jest";
import Vuex from "vuex";
import type { Link, RootState } from "../../../type";
import type { ConfigurationState } from "../../../store/configuration";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("LinkCellTitle", () => {
    it(`should render link title`, () => {
        const item = {
            id: 42,
            title: "my link",
            link_properties: {},
            type: TYPE_LINK,
        } as Link;

        const component_options = {
            localVue,
            propsData: {
                item,
            },
        };

        const configuration = { project_id: 101 } as unknown as ConfigurationState;
        const state = { configuration: configuration } as RootState;

        const store_options = {
            state,
        };

        const store = createStoreMock(store_options);
        const wrapper = shallowMount(LinkCellTitle, {
            mocks: { $store: store },
            ...component_options,
        });

        expect(wrapper.element).toMatchSnapshot();
    });
});
