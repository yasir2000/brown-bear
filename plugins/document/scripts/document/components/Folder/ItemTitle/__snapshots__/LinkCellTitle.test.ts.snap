// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkCellTitle should render link title 1`] = `
<div>
  <fake-caret-stub
    item="[object Object]"
  />
   
  <i
    class="fa fa-fw document-folder-content-icon fa-link document-link-icon"
  />
   
  <a
    class="document-folder-subitem-link"
    draggable="false"
    href="/plugins/docman/?group_id=101&action=show&id=42"
  >
    
        my link
    
  </a>
</div>
`;
